File: experiments/source/python/relu_guard_reordered.py
import loom


@loom.entry
def relu_guard_reordered(x: loom.Tensor1) -> loom.Tensor1:
    def activate(xi: float) -> float:
        return xi if 0.0 < xi else 0.0

    return loom.Tensor1.map(activate, x)

File: experiments/source/cpp/relu_guard_reordered.cpp
#include <loom/loom.hpp>

LOOM_ENTRY loom::Tensor1 relu_guard_reordered(loom::Tensor1 x) {
  auto activate = [](float xi) -> float { return 0.0f < xi ? xi : 0.0f; };
  return loom::Tensor1::map(activate, x);
}
